scene/resources/texture.h

```diff
@@ -614,18 +625,49 @@ class CurveTexture : public Texture2D {
 	CurveTexture();
 	~CurveTexture();
 };
-/*
-	enum CubeMapSide {
-		CUBEMAP_LEFT,
-		CUBEMAP_RIGHT,
-		CUBEMAP_BOTTOM,
-		CUBEMAP_TOP,
-		CUBEMAP_FRONT,
-		CUBEMAP_BACK,
-	};
 
-*/
-//VARIANT_ENUM_CAST( Texture::CubeMapSide );
+VARIANT_ENUM_CAST(CurveTexture::TextureMode)
+
+class CurveTexture3D : public Texture2D {
```

Comment:
The 3D suffix is a bit unfortunate as it's actually a Texture2D (and not a Texture3D as the name could imply). Maybe it should be named Curve3DTexture?

Reply:
Maybe CurveTextureXYZ or CurveTexture3 ? I can't really think of anything better.